core/utils.py

```diff
@@ -369,9 +369,11 @@ def tryint(x):
         return x
 
 
-def get_top_hoisted_role(member: discord.Member):
+def get_top_role(member: discord.Member, hoisted=True):
     roles = sorted(member.roles, key=lambda r: r.position, reverse=True)
     for role in roles:
+        if not hoisted:
+            return role
```

Comment:
Here you can just do:
```py
    if not hoisted:
        return member.top_role
```
before the `for` loop. Hmm, but the output still the same tho. 😅

And also, additionally, I'm suggesting if `hoisted` is `True` and no hoisted role is found after the `for` loop, it should fallback to `member.top_role`. Just to make sure this function return an appropriate role object.

Reply:
> Here you can just do:

Thanks for the recommendation. It's a bit of a nitpick perhaps, as in discord.py, it's also iterating over the roles anyway in a similar manner. I'm inclined to leave it as is.

>  I'm suggesting if hoisted is True and no hoisted role is found after the for loop, it should fallback to member.top_role. Just to make sure this function returns an appropriate role object. 

Honestly, this should probably have been the default behavior back when the original PR change occurred, however I didn't want to assume there wasn't considerations behind why it wasn't. Instead, my intention is to not alter the existing behavior as was written by @fourjr originally, and just make it configurable. 

I don't want to creep the scope of this PR beyond this intention. If I did, I'd probably end up re-writing the entire project eventually just because I found things a whole bunch of things I think should be different. 

If the behavior you state is something you want, feel free to create a PR after this is merged.